src/__tests__/variables.test-d.ts

```diff
@@ -36,6 +36,23 @@ describe('getVariablesType', () => {
     }>();
   });
 
+  it('works for one-of-objects', () => {
+    const query = `
+      mutation ($id: ID!, $input: OneOfPayload!) {
+        toggleTodo (id: $id input: $input) { id }
+      }
+    `;
+
+    type doc = parseDocument<typeof query>;
+    type variables = getVariablesType<doc, schema>;
+
+    expectTypeOf<variables>().toEqualTypeOf<{
+      id: string;
+      // TODO: should we get rid of the leading empty object?
```

Comment:
Yes, as per the RFC, exactly one property must be defined, i.e. all values are required and one must be specified when the input is passed